src/ast/check_shadowing.rs

```diff
@@ -1,18 +1,46 @@
-use crate::*;
+use crate::{util::HashMap, *};
 
 #[derive(Clone, Debug, Default)]
-pub(crate) struct Names(HashMap<String, Span>);
+pub(crate) struct Names {
+    seen: HashMap<String, Span>,
+    global_aliases: HashMap<String, (String, Span)>,
+}
 
 impl Names {
     fn check(&mut self, name: String, new: Span) -> Result<(), Error> {
-        if let Some(old) = self.0.get(&name) {
+        if let Some(old) = self.seen.get(&name) {
             Err(Error::Shadowing(name, old.clone(), new))
         } else {
-            self.0.insert(name, new);
+            self.seen.insert(name, new);
             Ok(())
         }
     }
 
+    fn track_global_alias(&mut self, name: &str, span: &Span) {
+        if let Some(stripped) = name.strip_prefix(GLOBAL_NAME_PREFIX) {
```

Comment:
Why do we only track globals with the prefix? What if we have the warning on, then wouldn't we have globals without the prefix?

Reply:
Without the prefix we can't tell if the user intended a global or not in a query variable, so we can't give an error. global_aliases tracks the ones we can error for